### PR TITLE
debian,rpm: add sudoers.d file to allow ceph-osd to run smartctl -x --json $device

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: sudoers.d/ceph-osd-smartctl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## allow ceph-osd (which runs as user ceph) to collect device health metrics
+
+ceph ALL=NOPASSWD: /usr/sbin/smartctl -a --json /dev/*
```
Comment (ErwanAliasr1) [inline sudoers.d/ceph-osd-smartctl]: I'm not sure about the need of specifying the arguments.
I tried locally and this syntax doesn't prevent from running any other smartctl options.

Comment (liewegas) [reply sudoers.d/ceph-osd-smartctl]: Can you retest?  On my system (fedora 28) I verified that the given arguments were required and that the /dev/* didn't match e.g. /dev/*/*.  I'm not sure I verified new arguments couldn't be added at the end, though, but I don't think smartctl can do anything dangerous there.

Comment (ErwanAliasr1) [reply sudoers.d/ceph-osd-smartctl]: You are perfectly true. I add another sudoers overriding that one :(

**Comment by ErwanAliasr1** [comment]
Just noticed that smartmontools is neither a requires. Is it optional or mandatory ?

**Comment by liewegas** [comment]
Good point.  We could make it a recommends instead of requires since we don't really need it for the OSD to function, but I wonder if that's the best path.  @ktdreyer @b-ranto ?

**Comment by liewegas** [comment]
Based on that doc I wonder if the Recommends: is more appropriate
```
This declares a strong, but not absolute, dependency.

The Recommends field should list packages that would be found together with this one in all but unusual installations)
```

**Comment by liewegas** [comment]
I don't think it makes sense to add the dependency yet, though, until we have an upstream version to require, since we depend on the --json output mode.

**Pull request merged**